set.mm

```diff
@@ -147746,6 +147746,473 @@ of an open range of nonnegative integers (of length equal to the length of
   $}
 
 
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Prefixes of a word
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+
+  In ~ https://www.allacronyms.com/prefix/abbreviated , "pfx" is proposed as
+  abbreviation for "prefix".  Regarding the meaning of "prefix", it is
+  different in computer science (automata theory/formal languages) compared
+  with linguistics: in linguistics, a prefix has a meaning (see Wikipedia
+  "Prefix" ~ https://en.wikipedia.org/wiki/Prefix ), whereas in computer
+  science, a prefix is an arbitrary substring/subword starting at the
+  beginning of a string/word (see Wikipedia "Substring"
+  ~ https://en.wikipedia.org/wiki/Substring#Prefix or
+  ~ https://math.stackexchange.com/questions/2190559/
+  is-there-standard-terminology-notation-for-the-prefix-of-a-word ).
+
+$)
+
+  $c prefix $.
+
+  $( Syntax for the prefix operator. $)
+  cpfx $a class prefix $.
+
+  ${
+    $d l s $.
+    $( Define an operation which extracts prefixes of words, i.e. subwords
+       starting at the beginning of a word.  Definition in section 9.1 of
+       [AhoHopUll] p. 318.  "pfx" is used as label fragment.  (Contributed by
+       AV, 2-May-2020.) $)
+    df-pfx $a |- prefix = ( s e. _V , l e. NN0
+                               |-> ( s substr <. 0 , l >. ) ) $.
+  $}
+
+  ${
+    $d L l s $.  $d S l s $.  $d V l s $.
+    $( Value of a prefix.  (Contributed by AV, 2-May-2020.) $)
+    pfxval $p |- ( ( S e. V /\ L e. NN0 )
+                    -> ( S prefix L ) = ( S substr <. 0 , L >. ) ) $=
+      ( vs vl wcel cn0 wa cvv cv cc0 cop csubstr co cpfx wceq df-pfx a1i adantl
+      cmpt2 simpl opeq2 oveq12d elex adantr simpr ovexd ovmpt2d ) ACFZBGFZHZDEA
+      BIGDJZKEJZLZMNZAKBLZMNZOIODEIGUOTPUKDEQRULAPZUMBPZHZUOUQPUKUTULAUNUPMURUS
+      UAUSUNUPPURUMBKUBSUCSUIAIFUJACUDUEUIUJUFUKAUPMUGUH $.
+
+    $( A zero length prefix.  (Contributed by AV, 2-May-2020.) $)
```

Comment:
The zero length prefix is the empty set.

Reply:
Done